system/core/util/Util.cfc

```diff
@@ -94,6 +101,29 @@ Description :
 		</cfscript>
 	</cffunction>
 
+	<!--- threadWithinThreadSupported --->
+	<cffunction name="threadWithinThreadSupported" access="public" returntype="boolean" hint="I return whether or not the current engine allows spawning threads from child threads" output="false">
+		<cfreturn variables.threadWithinThreadSupported />
+	</cffunction>
+
+	<!--- checkThreadWithinThreadSupport --->
+	<cffunction name="checkThreadWithinThreadSupport" access="private" returntype="any" output="false" hint="A little test logic to check for thread within thread support">
```

Comment:
> Couldn't we just do this by engine detection, or is it a specific version of Lucee that introduced this?

My thinking was that, should ACF not do this now, but change it in the future, it would just work. It is a specific version of Lucee that fixed it.